src/langcheck/metrics/eval_clients/_gemini.py

```diff
@@ -0,0 +1,242 @@
+                    f"Gemini's safety settings blocked the {i}th prompt:\n {response.candidates[0].safety_ratings}"  # NOQA: E501
+                )
+                responses[i] = None
+        return responses
+
+    def get_text_responses(
+            self,
+            prompts: Iterable[str],
+            *,
+            tqdm_description: str | None = None) -> list[str | None]:
+        '''The function that gets resonses to the given prompt texts.
+
+        Args:
+            prompts: The prompts you want to get the responses for.
+        Returns:
+            A list of responses to the prompts. The responses can be None if the
+            evaluation fails.
+        '''
+        config = {"generation_config": {"temperature": 0.0}}
+        config.update(self._generate_content_args or {})
+        tqdm_description = tqdm_description or 'Intermediate assessments (1/2)'  # NOQA: E501
+        responses = self._call_api(prompts=prompts,
+                                   config=config,
+                                   tqdm_description=tqdm_description)
+        response_texts = [
+            response.text if response else None for response in responses
+        ]
+
+        return response_texts
+
+    def get_float_score(
+            self,
+            metric_name: str,
+            language: str,
+            unstructured_assessment_result: list[str | None],
+            score_map: dict[str, float],
+            *,
+            tqdm_description: str | None = None) -> list[float | None]:
+        '''The function that transforms the unstructured assessments (i.e. long
+        texts that describe the evaluation results) into scores. We leverage the
+        function calling API to extract the short assessment results from the
+        unstructured assessments, so please make sure that the model you use
+        supports function calling
+        (https://ai.google.dev/gemini-api/docs/function-calling#supported-models).
+
+        Ref:
+            https://ai.google.dev/gemini-api/docs/function-calling
+
+        Args:
+            metric_name: The name of the metric to be used. (e.g. "toxicity")
+            language: The language of the prompts. (e.g. "en")
+            unstructured_assessment_result: The unstructured assessment results
+                for the given assessment prompts.
+            score_map: The mapping from the short assessment results
+                (e.g. "Good") to the scores.
+            tqdm_description: The description to be shown in the tqdm bar.
+
+        Returns:
+            A list of scores for the given prompts. The scores can be None if
+            the evaluation fails.
+        '''
+        if language not in ['en', 'ja', 'de', 'zh']:
+            raise ValueError(f'Unsupported language: {language}')
+
+        # Gemini's function calling API is similar to OpenAI's, so we can reuse
+        # the prompt template.
+        fn_call_template = get_template(f'{language}/get_score/openai.j2')
```

Comment:
I think it is better to rename the prompts to `function_calling.j2` and `plain_text.j2` now.. What do you think?

Reply:
Yep good point! Renamed the files